File: Easy/Python/896. Monotonic Array.py
class Solution(object):
    def isMonotonic(self, nums):
        if len(nums) == 1:
            return True
        
        inc = True
        dec = True

        for i in range(len(nums) - 1):
            if nums[i] < nums[i + 1]:
                dec = False
            if nums[i + 1] < nums[i]:
                inc = False
            if not inc and not dec:
                return False
        
        return inc or dec

File: Easy/Java/896. Monotonic Array.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        if(nums.length == 1)
            return true;
        
        boolean inc = true;
        boolean dec = true;

        for(int i = 0; i < nums.length - 1; i++){
            if(nums[i] < nums[i + 1])
                dec = false;
            if(nums[i + 1] < nums[i])
                inc = false;
            if(!inc && !dec)
                return false;
        }

        return inc || dec;
    }
}
